bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/userdefinedstate/UserStateHandler.java

```diff
@@ -0,0 +1,137 @@
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.boschshc.internal.devices.userdefinedstate;
+
+import static org.openhab.binding.boschshc.internal.devices.BoschSHCBindingConstants.CHANNEL_USER_DEFINED_STATE;
+
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.boschshc.internal.devices.BoschSHCConfiguration;
+import org.openhab.binding.boschshc.internal.devices.BoschSHCHandler;
+import org.openhab.binding.boschshc.internal.exceptions.BoschSHCException;
+import org.openhab.binding.boschshc.internal.services.userstate.UserStateService;
+import org.openhab.binding.boschshc.internal.services.userstate.dto.UserStateServiceState;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonElement;
+
+/**
+ * Handler for user defined states
+ *
+ * @author Patrick Gell - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class UserStateHandler extends BoschSHCHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private final UserStateService userStateService;
+    /**
+     * Bosch SHC configuration loaded from openHAB configuration.
+     */
+    private @Nullable BoschSHCConfiguration config;
+
+    public UserStateHandler(Thing thing) {
+        super(thing);
+
+        userStateService = new UserStateService();
+    }
+
+    @Override
+    public void initialize() {
+        var localConfig = this.config = getConfigAs(BoschSHCConfiguration.class);
+        String stateId = localConfig.id;
+        if (stateId == null || stateId.isBlank()) {
+            this.updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.conf-error.empty-state-id");
+            return;
+        }
+
+        // Try to get state info to make sure the state exists
+        try {
+            var bridgeHandler = this.getBridgeHandler();
+            var info = bridgeHandler.getUserStateInfo(stateId);
+            logger.trace("User-defined state initialized:\n{}", info);
+        } catch (TimeoutException | ExecutionException | BoschSHCException e) {
+            this.updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            return;
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            this.updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            return;
+        }
+        super.initialize();
```

Comment:
maybe we could call `super.initialize()` at the beginning of the method, or is there a specific reason to do it at the end?

Reply:
I have done the implementation like it was in the `BoschSHCDeviceHandler`. There is the `super.initialize()` also at the end of the method.
It would also work if I put the initialize code in the `initializeServices()` method.

If I put the `super.initialize()` call at the beginning of the method than the servies